l5kit/l5kit/evaluation/extract_losses.py

```diff
@@ -62,10 +69,18 @@ def compute_mse(A: np.ndarray, B: np.ndarray) -> np.ndarray:
         return ((A - B) ** 2).mean(axis=-1)  # reduce coords, keep steps
 
     errors = []
+    gt_avails = []
     for key, ground_truth_value in ground_truth.items():
-        errors.append(compute_mse(ground_truth_value, inference[key]))
+        gt_coords = ground_truth_value["coords"]
+        pred_coords = inference[key]["coords"]
+
+        errors.append(compute_mse(gt_coords, pred_coords))
+        gt_avails.append(ground_truth_value["avail"])
 
-    return np.array(errors).mean(axis=0)  # reduce samples, keep steps
+    # use numpy masked package to get a mean where unavailable elements are masked out
+    mask = ~np.array(gt_avails).astype(np.bool)  # note the ~ here, masked seems to assume 0 means keep
+    errors = np.ma.masked_array(np.array(errors), mask=mask)
```

Comment:
Why use np masked instead of simply errors[mask]?

Reply:
because the result of that cannot be reshaped (it's a flatten array) to keep the `future_step` axis, and we want to provide a value for each timestep